src/newHome/homeOnboarding/onboardingContext/useOnboardingStorage.ts

```diff
@@ -0,0 +1,85 @@
+import { useUser } from "@dashboard/auth";
+import { MetadataInput, useUpdateMetadataMutation } from "@dashboard/graphql";
+import {
+  OnboardingState,
+  StorageService,
+} from "@dashboard/newHome/homeOnboarding/onboardingContext/types";
+import debounce from "lodash/debounce";
+import { useCallback, useMemo } from "react";
+
+const METADATA_KEY = "onboarding";
+
+export const useOnboardingStorage = (): StorageService => {
+  const { user } = useUser();
+  const [updateMetadata] = useUpdateMetadataMutation({});
+
+  const getOnboardingState: StorageService["getOnboardingState"] = () => {
+    try {
+      const metadata = user?.metadata.find(m => m.key === METADATA_KEY);
+
+      if (!metadata) {
+        return undefined;
+      }
+
+      return JSON.parse(metadata.value);
+    } catch (error) {
+      // eslint-disable-next-line no-console
+      console.warn("Could not get onboarding state from metadata", { error });
+
+      return undefined;
+    }
+  };
+
+  const saveOnboardingState: StorageService["saveOnboardingState"] = useCallback(
+    async (onboardingState: OnboardingState) => {
+      if (!user) {
+        return;
+      }
+
+      try {
+        const userMetadata: Array<MetadataInput> = [
+          ...(user?.metadata?.map(data => ({
+            key: data.key,
+            value: data.value,
+          })) ?? []),
+        ];
+        const metadataValue = JSON.stringify(onboardingState);
+        const metadataIndex = userMetadata.findIndex(m => m.key === METADATA_KEY);
+
+        if (metadataIndex !== -1) {
+          userMetadata[metadataIndex] = {
+            key: METADATA_KEY,
+            value: metadataValue,
+          };
+        } else {
+          userMetadata.push({
+            key: METADATA_KEY,
+            value: metadataValue,
+          });
+        }
+
+        await updateMetadata({
+          variables: {
+            id: user?.id ?? "",
```

Comment:
Suggestion: We're already returning early if user is not present so `user.id` should be fine here I think